### DISCUSSION TITLE
How to access URL params from the context of a vscode extension?

### DISCUSSION THREAD

**Discussion by protozoo** [coder/code-server#6638] [Q&A]
I am developing a vscode extension intended to run in the browser with codeserver. I need to access the browser's url params (window.location.search). Specifically, I need to read the "folder" param. the vscode module doesn't seem to expose any relevant object for that purpose. Any hints on this? Thanks in advance

**Comment by code-asher** [comment]
code-server puts the last used query params in `~/.local/share/code-server/coder.json` so you could read that but it is not an official API.

However if you just need the current folder/workspace(s) you could try https://code.visualstudio.com/api/references/vscode-api#workspace.workspaceFolders

**Reply by protozoo** [reply]
thanks a lot for your reply @code-asher. Unfortunately I don't seem to have a code.json file in that location, I only see a folder named "coder-logs" with a couple of log files. 

I am now using the workspaceFolders as you suggested, which I think works well enough for my use case, although it can be problematic when a user adds more folders to the workspace (I am only interested in the "main" one, meaning the one being set by the "folder" query param.

**Reply by code-asher** [reply]
Oh weird, `coder.json` should be there since v4.0.1 I believe.  It defaults to the data directory, do you have it set to something else?  If not then I am not sure where it could be.  It might not exist until code-server is accessed via the browser at least once.

Thinking about it more though, using `folder` will not work anyway.  There are two query parameters that determine what VS Code will open: `folder` and `workspace`.  `folder` always means one folder and `workspace` means one or more.

1. When the user clicks "add folder to workspace" VS Code redirects, removing `folder` and adding `workspace`, so there would not even be a `folder` to use.
2. When opening a `workspace` there is no way to tell which the user thinks is most important so you probably have to handle them all.
3. In the case where both `folder` and `workspace` exist in the URL (the only way is if the user types this manually) then it looks like VS Code will take the last one, so even if `folder` is set it might not even be what is actually opened in the editor, since it might not be part of `workspace`.

**Reply by protozoo** [reply]
I see... Ok no prob, I think our current implementation works for our use case by using folder naming convention. Thanks a lot for your help, it's much appreciated :)
